Gordon360/Services/ScheduleService.cs

```diff
@@ -24,18 +24,25 @@ public ScheduleService(CCTContext context)
         /// Fetch the schedule item whose id and session code is specified by the parameter
         /// </summary>
         /// <param name="username">The AD Username of the student</param>
+        /// <param name="sessionID">The selected session of the student</param>
         /// <returns>StudentScheduleViewModel if found, null if not found</returns>
-        public async Task<IEnumerable<ScheduleViewModel>> GetScheduleStudentAsync(string username)
+        public async Task<IEnumerable<ScheduleViewModel>> GetScheduleStudentAsync(string username, string? sessionID)
         {
             var account = _context.ACCOUNT.FirstOrDefault(x => x.AD_Username == username);
 
             if (account == null)
             {
-                throw new ResourceNotFoundException() { ExceptionMessage = "The Schedule was not found." };
+                throw new ResourceNotFoundException() { ExceptionMessage = "The Account was not found." };
             }
 
-
-            var sessionCode = Helpers.GetCurrentSession(_context);
+            var sessionCode = "";
+            if (sessionID is not null)
+            {
+                sessionCode = sessionID;
+            } else
+            {
+                sessionCode = Helpers.GetCurrentSession(_context);
+            }
```

Comment:
There's a simpler way to express this using [the null coalescing operator](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/operators/null-coalescing-operator):
```suggestion
            var sessionCode = sessionID ?? Helpers.GetCurrentSession(_context);
```

Reply:
Make sure to update this in the faculty method as well.